### PR TITLE
JoinedUnderscore

### LINKED ISSUE #2422: AssembleMojoTest.java:45-51: Create new object that will...
The puzzle `1602-4a7b3be6` from #1602 has to be resolved: 

https://github.com/objectionary/eo/blob/c84b8c83f8725af850e01ac896fe13308070df85/eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java#L45-L51

The puzzle was created by maxonfjvipon on 18-Aug-23. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by h1alexbel**
closes #2422

---

## PR-Codex overview
This PR focuses on making changes to the code in the eo-maven-plugin module. 

### Detailed summary
- Added import statement for `org.eolang.maven.util.JoinedUnderscore` in `Place.java`
- Added `@Disabled` annotation to `OptimizeMojoTest.java` and `OptCachedTest.java`
- Added `JoinedUnderscoreTest.java` and `JoinedUnderscore.java` files
- Made changes to `AssembleMojoTest.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by h1alexbel** [comment]
@maxonfjvipon take a look, please, I found here unstable test, so that's why its disabled + puzzled

**Review by maxonfjvipon — COMMENTED** [review]
@h1alexbel looks good, but the puzzle assumes that code duplication also is resolved in `Place` class

File: eo-maven-plugin/src/main/java/org/eolang/maven/Place.java [hunk 1]
```diff
@@ -67,8 +68,10 @@ public Path make(final Path dir, final String ext) {
         out.append(this.name.title().replace(".", File.separator));
         this.name.label().ifPresent(
             version -> {
-                out.append('_');
-                out.append(version);
+                new JoinedUnderscore(
+                    out.append('_').toString(),
+                    out.append(version).toString()
+                ).asString();
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/main/java/org/eolang/maven/Place.java:74]: @h1alexbel I think you missed the idea here. You append `_` to `out`, then append `version`, then pass `_` and `version` to `JoinedUnderscore` and never use the result of it. 
Btw I think the result would be something like this: `__version`

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

File: eo-maven-plugin/src/main/java/org/eolang/maven/Place.java (hunk 1 above)
Comment (h1alexbel) [reply eo-maven-plugin/src/main/java/org/eolang/maven/Place.java:74]: I'm not sure, but I think I need to combine "_" and `version` altogether, and then add it to the `out`

Comment (h1alexbel) [reply eo-maven-plugin/src/main/java/org/eolang/maven/Place.java:74]:
```java
out.append(new JoinedUnderscore(version).asString());
```

**Comment by h1alexbel** [comment]
@maxonfjvipon could you check once again?

**Review by maxonfjvipon — APPROVED** [review]
@h1alexbel looks good to me, thanks
@yegor256 please have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**